jbpm-flow/src/main/java/org/jbpm/process/core/datatype/DataType.java

```diff
@@ -37,4 +37,14 @@
      */
     String getStringType();
 
+    /**
+     * Tries to convert parameter to an object of this type
+     * @param value to be converted. 
+     * For the sake of performance, it is assumed to not be null and not of <code>this</code> data type. 
+     * @return object converted to this type, if not possible to convert, the same object passed as parameter
+     */
+    default Object convertTo(Object value) {
+        return value;
```

Comment:
if we are going to do this, let's try to refine it a bit. convertTo is not really a good name. Would be better naming something like convertFrom  -> value is the source not the target.

All object primitive types has the valueOf method, so it would not be that bad to

Class -> get the class from String type
method = get the method valueOf  with argument value.getClass()
if the method is not found you can throw an error
method.invoke(class, valueOF, argument)

this is more generic and will allow more fine grain control even for user types.
Also this is generic so you can remove the impl from the sub types.

Reply:
Lets reduce the scope to be less ambition, I will change the method signature to valueOf(String value) and override it only for those types which implementation is trivial